server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -113,6 +113,7 @@ public final class ThreadContext implements Writeable {
     private static final ThreadContextStruct DEFAULT_CONTEXT = new ThreadContextStruct();
     private final Map<String, String> defaultHeader;
     private final ThreadLocal<ThreadContextStruct> threadLocal;
+    private final ThreadLocal<ThreadContextStruct> durableThreadLocal;
```

Comment:
@cwperks I don't think we need another `ThreadLocal` for that, please correct me if I am wrong but:
 - we could introduce `persistentHeaders` to `ThreadContextStruct` (similarly to `transientHeaders`) but: `persistentHeaders` should always be carried over and never stashed.

Reply:
Thank you @reta, I updated this now to add persistent headers and persistent transient headers to the ThreadContextStruct to remote multiple thread locals.

Reply:
Thanks @cwperks , what are these `persistentTransient` headers? I think the `persistentHeaders` would be sufficient, do you have a use case for it?

Reply:
@reta Similar to the non-persistent headers, there are 2 data structures depending on the data type:

header == String
transient == Object <- Object is general and to be transmitted Node <-> Node the transients need to be serializable

I created analogs of both of these, but I think from the security perspective its helpful to have persistent transients since the authenticated user is stored in a transient header: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/auth/BackendRegistry.java#L361-L364

I would like to write the User into a persistent transient header so that it is available regardless if the current threadcontext is stashed or not. More information can be found here: https://github.com/opensearch-project/security/issues/2909

Reply:
I like a generic `putPersistent` and `getPersistent` that mirrors how transients are currently implemented but makes them unstashable. I will update this PR.

Reply:
Updated to a single `putPersistent` and `getPersistent`. Its modeled after the transient headers except that they are unstashable.